### DISCUSSION TITLE
Deploying files to s3 using local api routes

### DISCUSSION THREAD

**Discussion by rohitmi0023** [vercel/next.js#20426] [Help]
So basically I want to deploy files from the frontend to amazon s3 buckets using nextjs local api(like which is inside `pages` directory) directory and store the file name to my database. I have done uploading files part in the past in development mode only using `multer` and `express`. But here, I am afraid to so because I saw that `vercel` doesn't supports custom server like express. I have searched here and there to get through this but still didn't get satisfactory answers. So I will highly appreciate if you guys help me through this.

**Comment by ijjk** [comment]
Hi, you are still able to leverage `express` middleware on Vercel using API routes, we have an example in our docs for this here https://nextjs.org/docs/api-routes/api-middlewares#connectexpress-middleware-support

**Reply by rohitmi0023** [reply]
@ijjk sorry for the late reply, I tried using middleware method like you suggested but I guess I have done done something wrong in the implementation part. I have used `multerS3` package in additon to multer.
```js
import aws from 'aws-sdk';
var Multer = require('multer');
var multerS3 = require('multer-s3');

const s3 = new aws.S3({
	accessKeyId: 'accesskey',
	secretAccessKey: `somesecret`,
});

const multer = Multer({
	storage: multerS3({
		s3: s3,
		bucket: 'some-bucket',
		metadata: function (req, file, cb) {
			cb(null, { fieldName: file.fieldname });
		},
		key: function (req, file, cb) {
			cb(null, Date.now().toString());
		},
	}),
}).single('subtitle');

function runMiddleware(req, res, fn) {
	return new Promise((resolve, reject) => {
		fn(req, res, result => {
			console.log(result);
			console.log(req);
			if (result instanceof Error) {
				return reject(result);
			}
			return resolve(result);
		});
	});
}

async function handler(req, res) {
	if (req.method === 'POST') {
		const response = await runMiddleware(req, res, multer);
		console.log(response);
		res.json({ message: 'Hello subtiltles!' });
	}
}

export default handler;
```
I get `undefined` for the last console message as well files doesn't gets uploaded too.
